Time-Card/DRV/ptp_ocp.c

```diff
@@ -2198,6 +2309,60 @@ ptp_ocp_mro50_ioctl(struct file *file, unsigned cmd, unsigned long arg)
 		iowrite32(MRO50_OP_SAVE_COARSE, &bp->osc->ctrl);
 		mutex_unlock(&bp->mutex);
 		return 0;
+	case ART_CALIBRATION_READ_PARAMETERS:
+		if (!bp->i2c_ctrl)
+			return -EFAULT;
+		/* Fetch calibration parameters from EEPROM 4th bloc of 256 bytes*/
+		dev = device_find_child(&bp->i2c_ctrl->dev, NULL, ptp_ocp_firstchild);
+		if (!dev) {
+			dev_err(&bp->pdev->dev, "Can't find I2C adapter\n");
+			return -EFAULT;
+		}
+
+		adap = i2c_verify_adapter(dev);
+		if (!adap) {
+			dev_err(&bp->pdev->dev, "device '%s' isn't an I2C adapter\n",
+				dev_name(dev));
+			return -EFAULT;
+		}
```

Comment:
this code appears several times in this commit, and also through out the driver. looks like a good opportunity to generalize the code into function

Reply:
I agree we should factorize.

How do you see the propotype of the function ?
We would need both adap and dev in order to correctly call put_device(dev);

How about static int ptp_ocp_get_i2c_adapter(struct ptp_ocp *bp,  struct device *dev, struct i2c_adapter *adap) ?

That way we could have all we need